### PR TITLE
Adding NCAR/vapor python package

### PR DESCRIPTION
**Pull request by StasJ**
Checklist
- [X] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [X] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [X] Source is from official source.
- [X] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [X] If static libraries are linked in, the license of the static library is packaged.
- [X] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [X] Build number is 0.
- [X] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [ ] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [X] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by sgpearse** [comment]
To whom it may concern: I am a developer on the NCAR/VAPOR project and am eager to support this important utility on Conda's infrastructure.

**Comment by StasJ** [comment]
@conda-forge-admin, please ping conda-forge/help-c-cpp

The linux build is failing with multiple of the following errors:

```
ERROR (vapor,lib/librender.so): $RPATH/libfreetype.so.6 not found in packages, sysroot(s) nor the missing_dso_whitelist.\n.. is this binary repackaging?
```

However all of the libraries the errors reference are in the required packages.

meta.yaml +25:
```
requirements:
   build:
     [...]
     - freetype =2.10.4
```

File: recipes/vapor/build.sh [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+
+unzip -d include buildutils/GTE.zip
+
+if [ ! -d "build" ]; then
+    mkdir build
+fi
+cd build
+
+cmake .. \
+    -DCONDA_BUILD=ON \
```
Comment (carterbox) [inline recipes/vapor/build.sh:51]:
```suggestion
cmake .. \
    -DBUILD_SHARED_LIBS:BOOL=ON \
    -DCMAKE_BUILD_TYPE=Release \
    -DCONDA_BUILD=ON \
```

File: recipes/vapor/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+  - url: https://github.com/NCAR/VAPOR-Data/archive/refs/tags/1.0.tar.gz
+    sha256: 03ce8f52104ba63049646db224b9221a2802f68544e77aa3642115194e5bfec8
+    folder: vapor-data
+
+build:
+  number: 0
+  skip: True  # [win]
+  skip: True  # [py<39]
+  skip_compile_pyc:
+    - "*.py"
```
Comment (carterbox) [inline recipes/vapor/meta.yaml:20]: Why? This is all python files? Maybe limit this to a specific directory?

File: recipes/vapor/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,89 @@
+{% set name = "VAPOR" %}
```
Comment (carterbox) [inline recipes/vapor/meta.yaml:1]: You need to open a PR on bioconda to remove their package at the same time that this one is added to conda-forge if it is the same package.

**Review by carterbox — CHANGES_REQUESTED** [review]
Please add a `requirements/host` section.

The `requirements/build` section is for build tools like cmake, make, doxygen, and compilers. These tools are only needed during the build and not needed again at runtime.

The `requirements/host` section is for libraries that target platform specific and that are needed at both build/compile time and again at run/link time.

Please move libraries need at compile time to the host section. i.e. numpy, jpeg, libtiff, ...

**Comment by StasJ** [comment]
> Please add a `requirements/host` section.
> 
> The `requirements/build` section is for build tools like cmake, make, doxygen, and compilers. These tools are only needed during the build and not needed again at runtime.
> 
> The `requirements/host` section is for libraries that target platform specific and that are needed at both build/compile time and again at run/link time.
> 
> Please move libraries need at compile time to the host section. i.e. numpy, jpeg, libtiff, ...

@carterbox if I move the libraries to `requirements/host` like so:
```
requirements:
  build:
    - make
    - ...
  host:
    - udunits2 =2.2
    - netcdf4 =1.5
    - ...
  run:
    - py-opencv 
    - ...
```

I get the following errors:
```
WARNING (vapor,lib/libvdc.dylib): Needed DSO lib/libudunits2.0.dylib found in ['udunits2']
WARNING (vapor,lib/libvdc.dylib): .. but ['udunits2'] not in reqs/run, (i.e. it is overlinking) (likely) or a missing dependency (less likely)
WARNING (vapor,lib/libvdc.dylib): Needed DSO lib/libnetcdf.19.dylib found in ['libnetcdf']
WARNING (vapor,lib/libvdc.dylib): .. but ['libnetcdf'] not in reqs/run, (i.e. it is overlinking) (likely) or a missing dependency (less likely)
...
```

And then conda fails to install the dependencies with the package.

**Comment by carterbox** [comment]
If your binary links to those libraries, then are needed at runtime. If they are not needed, but your binary contains references/symbols of those libraries, then you are overlinking.

**Comment by StasJ** [comment]
> If your binary links to those libraries, then are needed at runtime. If they are not needed, but your binary contains references/symbols of those libraries, then you are overlinking.

My binary links to those libraries and they are needed at runtime, the issue is Conda does not install them, despite being listed in `host`, so they are not available at runtime.

**Pull request closed without merging**